### DISCUSSION TITLE
Ender 3 cooling fan PWM frequency editing

### DISCUSSION THREAD

**Discussion by I-wish-2** [mriscoc/Ender3V2S1#702] [Q&A] [answered]
Greetings,
I'm modifying and Ender 3 Pro to run with a Solvo 5 watt laser.  I need to know how to adjust the PWM frequency in the Configuration_adc.h file to increase the PWM Hz.  In the current Professional firmware the value is 7.87 Hz.  I need to configure for about 500 Hz for better for a higher burn resolution.
I'm looking at the data starting at line 594 or there about and I'm having a bit of a time understanding how to resolve the formula to achieve a higher frequency.
Does anyone have a though or a how to?
Best Regards
Doc'

**Comment by I-wish-2** [comment]
I found the following information on github.com/MarlinFirmware/Marlin/issues/20642.
I followed the steps that chagwell outlined. Those edits and a re-compile resulted in the PWM frequency increasing from 7.87 Hz to 50Hz.  I'm trying to edit the FAST_PWM_FAN_FREQUENCY in the Configurations_adv.h file but my edits are not working as the frequency remains at 50 Hz.  If anyone has some thoughts or edits I can do please let me know.   The edits so far are as follows:

Comment out FAN_SOFT_PWM in Configuration.h
Change FAN_MIN_PWM to 0 in Configuration_adv.h
Uncomment FAST_PWM_FAN in Configuration_adv.h
Change MF_TIMER_STEP from 5 to 4 in Marlin/src/HAL/STM32F1/timers.h
Comment out FAN_SOFT_PWM_REQUIRED in Marlin/src/pins/stm32f1/pins_CREALITY_V4.h
Change PA_0 from TIM2 to TIM5 in buildroot/share/PlatformIO/variants/MARLIN_F103Rx/PeripheralPins.c

These resulted in the PWM frequency increasing from 7.87 Hz to 50Hz.

The step below is the one I'm not having any success with.

You can optionally uncomment and modify FAST_PWM_FAN_FREQUENCY in Configuration_adv.h if you need value different than 1kHz.

Thanks for the read, please help if you can
Regards
Doc'

**Comment by I-wish-2** [comment]
Some additional testing notes:
From Control > Temperature > Fan >  set points 0 to 255
Set Points.......Frequency........Volts DC 
2   > 21..............100 Hz...............035 to .716
22 > 255............50 Hz..............739 to 4.923

These two edits were not clear as to editing the value or commenting it out and uncommenting the other entry  I tried both but no effect.

Change MF_TIMER_STEP from 5 to 4 in Marlin/src/HAL/STM32F1/timers.h
Change PA_0 from TIM2 to TIM5 in buildroot/share/PlatformIO/variants/MARLIN_F103Rx/PeripheralPins.c

Thanks for the read, please help if you can
Regards
Doc'

**Reply by I-wish-2** [reply]
Using the following edits I was able to increase the PWM of the parts cooling fan frequency from 7.87 Hz to 1KHz.  This allowed the PWM signal to drive the Solvo 5W laser correctly.  The edits are listed below:

The PWM firmware edits:

Marlin  >  configuration.h:  Your line numbers could be different !  The lines are shown with the edits.
Comment out Line 3247  // #define FAN_SOFT_PWM   ( add the // marks to the front of the line )
*****
Marlin  >  configuration_adv.h:  Your line numbers could be different !  The lines are shown with the edits.
Edit Line 585  #define FAN_MIN_PWM 0   ( change the value here to 0 )
Uncomment Line 613  #define FAST_PWM_FAN   ( remove the // marks from front of the line )
*****
Marlin  >  src  >  pins  >  stm32f1  >  pins_CREALITY_V4.h
Comment out Line 154  //#define FAN_SOFT_PWM_REQUIRED  ( add the // marks to the front of the line )
*****

ini  >  stn32f1.ini:   Your line numbers could be different !  The lines are shown with the edits.

	Line 130	 -DHAL_SD_MODULE_ENABLED
After         Line 131      -DSS_TIMER=4 -DTIMER_SERVO=TIM5
Add new   Line 132     -DSS_TIMER=4 -DTIMER_SERVO=TIM8)

Results:  PWM frequency will increase from 7.87 Hz to 1KHz

Regards
Doug

**Comment by mriscoc** [comment, marked as answer]
Sorry for not commenting here but I don't have a way to help you. I don't know if this is relevant to you:
https://github.com/CrealityOfficial/Ender-3S1/issues/46
https://github.com/mriscoc/Ender3V2S1/discussions/62
https://github.com/mriscoc/Ender3V2S1/discussions/477

**Reply by I-wish-2** [reply]
Greetings,
Thank you for your reply and regarding the PWM frequency the above edits allow me to move to the 1KHz frequency.  The laser is now operating fine although there is still some issue with getting the X and Y axis to not lag the laser during a burn.  It will burn lines just fine but not raster's or " fills " as LightBurn refers to it.
I do have an Ender3 S1 being delivered here today and after it is set up and printing ok I'm going to add the Creality CV 5W laser kit to it and start again.
Thank again and best regards
Doug

**Comment by mriscoc** [comment]
Some preliminary laser support is being discussed here: https://github.com/mriscoc/Ender3V2S1/discussions/62
